### PR TITLE
Automated release scripts

### PR DESCRIPTION
**Pull request by dnephin**
Some scripts I worked on to automate parts of the release (while working on the 1.4.1 release).

The last step (`push-release`) still needs some work to get the osx steps automated.

https://github.com/dnephin/compose/blob/release_scripts/project/RELEASE-PROCESS.md

### CODE REVIEW

**Comment by thaJeztah** [comment]
Nice! Something worth considering; "do not allow the release of a dirty binary"; https://github.com/docker/docker/pull/16224

**Comment by bfirsh** [comment]
Engine might also be interested in this! /cc @icecrime

**Comment by dnephin** [comment]
I think it would be awesome to generalize this, many of the concerns are the same (and the ones that aren't can be abstracted away).

I took care to put most of the compose-specific things into constants, so we can generalize it without too much effort.

**Comment by mnowster** [comment]
Nice work :sparkles:

File: project/RELEASE-PROCESS.md [hunk 1]
```diff
@@ -24,72 +24,36 @@ Building a Compose release
...
-1.  Check that CI is passing on the bump PR.
+## To release a version (whether RC or stable)
 
-2.  Check out the bump branch:
+Check out the bump branch and run the `build-binary` script
 
         git checkout bump-$VERSION
+        ./script/release/build-binary
 
-3.  Build the Linux binary:
```
Comment (mnowster) [inline project/RELEASE-PROCESS.md]: Given these steps aren't automated yet, can we keep the docs in for how to build mac/windows binaries in?

Comment (dnephin) [inline project/RELEASE-PROCESS.md]: good idea

**Comment by mnowster** [comment]
LGTM

**Pull request merged**